### PR TITLE
[bazel] Improve libc build on macOS

### PR DESCRIPTION
**Pull request by keith**
With these changes you can now `bazel build @llvm-project//...` on macOS, and the targets in libc that don't yet support macOS (or non-linux) are ignored

### CODE REVIEW

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel
Comment (nickdesaulniers) [inline utils/bazel/llvm-project-overlay/libc/BUILD.bazel:1008]: Note: I'm about to touch quick_exit in https://github.com/llvm/llvm-project/pull/85955. Not sure if that will cause a conflict here.

**Review by michaelrj-google — APPROVED** [review]
This patch LGTM, I will warn you that there hasn't been much work on the MacOS build (bazel or cmake) recently so there may be some broken pieces. Feel free to send issues my way, but also we could absolutely use your expertise in bringing it online.

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel
Comment (keith) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel:1008]: i can wait for yours, there's no rush on this one

**Comment by keith** [comment]
good to know thanks! I'm not sure how much more I will try to do, originally I just set out to do a `bazel build ...` and started dealing with the failures, but I'm not sure if we will have a dep on this that will require more fixes, we'll see!

**Comment by keith** [comment]
rebased and pushed some more changes so now nothing fails to build on macOS (but many things are just skipped)

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel
Comment (gchatelet) [inline utils/bazel/llvm-project-overlay/libc/BUILD.bazel:986]: Technically this condition should be `@platforms//os:linux` and an additional empty "//conditions:default" clause.

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel
Comment (gchatelet) [inline utils/bazel/llvm-project-overlay/libc/BUILD.bazel:1003]: same here

**Review by gchatelet — COMMENTED** [review]
LGTM modulo a few comments. I'd need to check that this is compatible with downstream before landing this change.

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel
Comment (keith) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel:986]: updated

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel
Comment (keith) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel]: updated

**Review by gchatelet — APPROVED** [review]
Downstream is happy, LGTM

**Review by lntue — APPROVED** [review]

**Pull request merged**
